source/brailleDisplayDrivers/ecoBraille.py

```diff
@@ -67,41 +69,53 @@ class ecoTypes:
 	TECO_40 = 40
 	TECO_80 = 80
 
-def eco_in_init(dev):
-	msg = dev.read(9)
-	if (len(msg) < 9):
+def eco_in_init(dev: serial.Serial) -> int:
+	msg: bytes = dev.read(9)
+	if len(msg) < 9:
 		return ecoTypes.TECO_80 # Needed to restart NVDA with Ecoplus
-	msg = struct.unpack('BBBBBBBBB', msg)
+	msgUnpacked: Tuple[int, ...] = struct.unpack(b'BBBBBBBBB', msg)
 	# Command message from EcoBraille is something like that:
 	# 0x10 0x02 TT AA BB CC DD 0x10 0x03
 	# where TT can be 0xF1 (identification message) or 0x88 (command pressed in the line)
 	# If TT = 0xF1, then the next byte (AA) give us the type of EcoBraille line (ECO 80, 40 or 20)
-	if (msg[0] == 0x10) and (msg[1] == 0x02) and (msg[7] == 0x10) and (msg[8] == 0x03):
-		if msg[2] == 0xf1: # Initial message
-			if (msg[3] == 0x80):
+	if (
+		(msgUnpacked[0] == 0x10)
+		and (msgUnpacked[1] == 0x02)
+		and (msgUnpacked[7] == 0x10)
+		and (msgUnpacked[8] == 0x03)
+	):
+		if msgUnpacked[2] == 0xf1:  # Initial message
+			if msgUnpacked[3] == 0x80:
 				return ecoTypes.TECO_80
-			if (msg[3] == 0x40):
+			if msgUnpacked[3] == 0x40:
 				return ecoTypes.TECO_40
-			if (msg[3] == 0x20):
+			if msgUnpacked[3] == 0x20:
 				return ecoTypes.TECO_20
 	return ecoTypes.TECO_80 # Needed for changing Braille Settings with Ecoplus
 
-def eco_in(dev):
-	msg = dev.read(9)
+def eco_in(dev: serial.Serial) -> int:
 	try:
-		msg = struct.unpack('BBBBBBBBB', msg)
+		msg: Tuple[int, ...] = struct.unpack(b'BBBBBBBBB', dev.read(9))
```

Comment:
Also here, indexing the bytes object just returns ints, so we no longer have to unpack.